### PR TITLE
changes dz_min from defined constant to a namelist option

### PR DESCRIPTION
**Pull request by bensonr**
Second step of the synchronization process for dev/emc branch within the NOAA-GFDL organization.

### CODE REVIEW

**Review by XiaqiongZhou-NOAA — APPROVED** [review]
It looks fine to me.
Just wondering why not merge the EMC fork to GFDL emc/dev since these are already included in the EMC fork. 
Should the description be added in fv_control.F90 as before ? " !> \param[in] dz\_min  Real: Minimum layer thickness to enforce monotonicity of height to prevent blowup. Set to 2 by default."

**Review by laurenchilutti — APPROVED** [review]

File: model/nh_utils.F90
Comment (junwang-noaa) [inline model/nh_utils.F90:199]: This is different from current EMC fork, where I think Kate switched back to k=km,1,-1 in her [PR#53](https://github.com/NOAA-EMC/GFDL_atmos_cubed_sphere/pull/53)
https://github.com/NOAA-EMC/GFDL_atmos_cubed_sphere/pull/53/files#diff-ab35d906facb07c24877b72bbc6b6acc3e856a82cb065a5a6d3391f3a149969aR200

File: model/nh_utils.F90
Comment (junwang-noaa) [inline model/nh_utils.F90:313]: Same here

**Comment by bensonr** [comment]
@XiaqiongZhou-NOAA - I didn't try using the existing PR because I was
concerned the different commit history might throw the synchronization
off.  Plus, I want the dz_min commit to come in by itself as there is a
possibility of it coming back into master.

File: model/nh_utils.F90
Comment (bensonr) [reply model/nh_utils.F90]: After discussions with @junwang-noaa, I have included the master version of the height monotonicity constraint for update_dz_*

**Comment by lharris4** [comment]
Hi, all. I believe that Xi and I had been discussing this earlier: for
height processes, you always want to go bottom-up.

Thanks,
Lucas

**Comment by junwang-noaa** [comment]
Yes, that (go bottom-up) is in the latest dev/emc branch at EMC fork, the
same as GFDL master branch

**Comment by bensonr** [comment]
I remember the discussion and am glad @junwang-noaa  brought it to my attention when reviewing the dz_min changes - even though it wasn't part of this particular PR.

File: model/nh_utils.F90
Comment (bensonr) [reply model/nh_utils.F90]: @junwang-noaa can you take one more look and approve if this is now good - thanks!

**Review by junwang-noaa — APPROVED** [review]
Looks good to me, thanks.

**Pull request merged**
